Lib/imaplib.py

```diff
@@ -1279,6 +1385,236 @@ def print_log(self):
+            imap._mesg('idle start'
+                       + ('' if self._dur is None else f' dur={self._dur}'))
+
+        try:
+            # Start capturing untagged responses before sending IDLE,
+            # so we can deliver via iteration any that arrive while
+            # the IDLE command continuation request is still pending.
+            imap._idle_capture = True
+
+            self._tag = imap._command('IDLE')
+            # Process responses until the server requests continuation
+            while resp := imap._get_response():  # Returns None on continuation
+                if imap.tagged_commands[self._tag]:
+                    raise imap.abort(f'unexpected status response: {resp}')
+
+            if __debug__ and imap.debug >= 4:
+                prompt = imap.continuation_response
+                imap._mesg(f'idle continuation prompt: {prompt}')
+        except:
+            imap._idle_capture = False
+            raise
+
+        self._sock_timeout = imap.sock.gettimeout() if imap.sock else None
+        if self._sock_timeout is not None:
+            imap.sock.settimeout(None)  # Socket timeout would break IDLE
+
+        self._old_state = imap.state
+        imap.state = 'IDLING'
+
+        return self
+
+    def __iter__(self):
+        return self
+
+    def _wait(self, timeout=None):
+        # Block until the next read operation should be attempted, either
+        # because data becomes availalable within 'timeout' seconds or
+        # because the OS cannot determine whether data is available.
+        # Return True when a blocking read() is worth trying
+        # Return False if the timeout expires while waiting
+
+        imap = self._imap
+        if timeout is None:
+            return True
+        if imap._readbuf:
+            return True
+        if timeout <= 0:
+            return False
+
+        if imap.sock:
+            fileobj = imap.sock
+        elif platform.system() == 'Windows':
+            return True  # Cannot select(); allow a possibly-blocking read
+        else:
+            fileobj = imap.readfile
+
+        if __debug__ and imap.debug >= 4:
+            imap._mesg(f'idle _wait select({timeout})')
+
+        with selectors.DefaultSelector() as sel:
+            sel.register(fileobj, selectors.EVENT_READ)
+            readables = sel.select(timeout)
+            return bool(readables)
+
+    def _pop(self, timeout, default=('', None)):
+        # Get the next response, or a default value on timeout
+        #
+        # :param timeout:   Time limit (in seconds) to wait for response
+        # :type timeout:    int|float|None
+        # :param default:   Value to return on timeout
+        #
+        # Note: This method ignores 'dur' in favor of the timeout argument.
+        #
+        # Note: Windows IMAP4_stream connections will ignore the timeout
+        # argument and block until the next response arrives, because
+        # Windows select() only works on sockets.
```

Comment:
The docstring here should be in a string, not a comment.
```suggestion
    def _pop(self, timeout, default=('', None)):
        """Get the next response, or a default value on timeout

        :param timeout:   Time limit (in seconds) to wait for response
        :type timeout:    int|float|None
        :param default:   Value to return on timeout

        Note: This method ignores 'dur' in favor of the timeout argument.

        Note: Windows IMAP4_stream connections will ignore the timeout
        argument and block until the next response arrives, because
        Windows select() only works on sockets."""
```

(I'm not sure why GitHub is making the diff look weird.)